scripts/travis-install-libgit2.sh

```diff
@@ -0,0 +1,14 @@
+#!/usr/bin/env bash
+set -o errexit -o nounset
+
+LIBGIT2_VERSION="0.27.4"
+
+mkdir libgit2
+
+wget https://github.com/libgit2/libgit2/archive/v${LIBGIT2_VERSION}.tar.gz
+tar -xf v${LIBGIT2_VERSION}.tar.gz -C ./libgit2 --strip-components=1
+
+cd libgit2
+mkdir build && cd build
+cmake .. -DCMAKE_INSTALL_PREFIX=../install -DBUILD_CLAR=OFF
+cmake --build . --target install
```

Comment:
Re-building it every time? That's going to slow down every test job :(

Reply:
I started looking at caching but it's quite hard to detect the current state of the cache and in case of build-failures we get broken caches that are even harder to detect.

The download and build takes ~10-18 seconds at most at the moment. I'm open to suggestions that make cache detection easier to implement :(

Reply:
I'm happy to do a follow-up once all that stuff is actually implemented and landed to look at improving build-times but that's time-intensive and my time-boxed experiments didn't bear any fruits